backend/deepchecks_monitoring/ee/bgtasks/object_storage_ingestor.py

```diff
@@ -72,91 +77,128 @@ async def run(self, task: 'Task', session: AsyncSession, resources_provider: Res
         model: Model = (await session.scalar(select(Model).options(selectinload(Model.versions))
                                              .where(Model.id == model_id)))
         if model is None:
-            await session.commit()
+            self._handle_error(errors, f'Could not locate model with id {model_id}')
+            await self._finalize_before_exit(session, errors)
             return
 
         # Get s3 authentication info
         s3_data_source = (await session.scalar(select(DataSource).where(DataSource.type == 's3')))
         if s3_data_source is None:
-            self.logger.warning({'message': 'No data source of type s3 found'})
-            await session.commit()
+            self._handle_error(errors, 'No data source of type s3 found', model_id, set_warning_in_logs=True)
```

Comment:
it will spam our db

Reply:
@JKL98ISR we can implement purge job to delete old errors later on if we will see that it's too much...